.github/workflows/My-CI-Pipeline.yml

```diff
@@ -16,4 +16,32 @@ jobs:
     - uses: actions/checkout@v2
     - run: docker build --tag dani --target test .
     - run: docker run --env-file .env.test dani
-     
\ No newline at end of file
+
+ build-two:
+    name: build production image and push to dockerhub  
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - if: ${{ github.event_name == 'push' && github.event.action == 'm8' }}
+        run: echo "publishing!"
```

Comment:
I missed this last time, but note that this only runs _this step_ if you're on the relevant branch - we want to actually limit the whole job (not pipeline) to the main branch. See an example of doing so below